docs/developer/architecture/core/patterns-scoped-services.asciidoc

```diff
@@ -0,0 +1,85 @@
+[[patterns]]
+== Patterns
+[[scoped-services]]
+=== Scoped services
+Whenever Kibana needs to get access to data saved in elasticsearch, it
+should perform a check whether an end-user has access to the data. In
+the legacy platform, Kibana requires binding elasticsearch related API
+with an incoming request to access elasticsearch service on behalf of a
+user.
+
+[source,js]
+----
+async function handler(req, res) {
+  const dataCluster = server.plugins.elasticsearch.getCluster('data');
+  const data = await dataCluster.callWithRequest(req, 'ping');
+}
+----
+
+The Kibana Platform introduced a handler interface on the server-side to perform that association
+internally. Core services, that require impersonation with an incoming
+request, are exposed via `context` argument of
+{kib-repo}blob/{branch}/docs/development/core/server/kibana-plugin-core-server.requesthandler.md[the
+request handler interface.] The above example looks in the Kibana Platform
+as
+
+[source,js]
+----
+async function handler(context, req, res) {
+  const data = await context.core.elasticsearch.client.asCurrentUser('ping');
```

Comment:
Let's remove legacy ES client API mentions from the docs. Devs should start using the new ES client instead.

Reply:
I restructured the section and removed the "legacy" way of doing things.